tests/zfs-tests/include/libtest.shlib

```diff
@@ -328,6 +328,38 @@ function create_bookmark
 	log_must zfs bookmark $fs_vol@$snap $fs_vol#$bkmark
 }
 
+#
+# Create a temporary clone result of an interrupted resumable 'zfs receive'
+# $1 Destination filesystem name. Must not exist, will be created as the result
+#    of this function along with its %recv temporary clone
+# $2 Source filesystem name. Must not exist, will be created and destroyed
+#
+function create_recv_clone
+{
+	typeset recvfs="$1"
+	typeset sendfs="${2:-$TESTPOOL/create_recv_clone}"
+	typeset snap="$sendfs@snap1"
+	typeset incr="$sendfs@snap2"
+	typeset mountpoint="$TESTDIR/create_recv_clone"
+	typeset sendfile="$TESTDIR/create_recv_clone.zsnap"
+
+	 [[ -z $recvfs ]] && log_fail "Recv filesystem's name is undefined."
```

Comment:
You could enforce `recvfs` doesn't exist by calling `datasetexists` and failing if that returns true. Likewise with `sendfs`, you'd want to make sure it doesn't exist.